compilerProvider.js

```diff
@@ -211,7 +211,7 @@ CompilerProvider.prototype.isLocal = function(localPath){
  * @return {String}          path
  */
 CompilerProvider.prototype.resolveCache = function(fileName){
-  const thunk = findCacheDir({name: 'truffle', thunk: true});
+  const thunk = findCacheDir({name: 'truffle', cwd: __dirname, thunk: true});
```

Comment:
Why's this necessary, just curious?

Reply:
The idea is to cache at the `node_modules` of the truffle instance invoked, so global truffle has a cache, truffle installed in local `node_modules` has it's own cache. The `find-cache-dir` doc says the default setting is `process.cwd` as the starting point of an outward search for a `package.json` e.g. a directory w/ a node_modules it can deposit a cache in. But a truffle project doesn't require a `package.json` so was worried this is a little non-deterministic for our case. Like what if it hits something on the way out, totally unrelated?

WDYT?

Reply:
@gnidan Merging but happy to revisit this fwiw.